src/abstractdataframe/reshape.jl

```diff
@@ -185,39 +193,72 @@ long = stack(wide)
 wide0 = unstack(long)
 wide1 = unstack(long, :variable, :value)
 wide2 = unstack(long, :id, :variable, :value)
+wide3 = unstack(long, [:id, :a], :variable, :value)
 ```
 Note that there are some differences between the widened results above.
-
 """
 function unstack(df::AbstractDataFrame, rowkey::Int, colkey::Int, value::Int)
-    # `rowkey` integer indicating which column to place along rows
-    # `colkey` integer indicating which column to place along column headers
-    # `value` integer indicating which column has values
-    refkeycol = CategoricalArray{Union{eltype(df[rowkey]), Missing}}(df[rowkey])
-    # make sure we report only levels actually present in rowkey column
-    # this is consistent with how the other unstack method based on groupby works
+    refkeycol = copy(CategoricalArray{Union{eltype(df[rowkey]), Missing}}(df[rowkey]))
```

Comment:
I don't think you need `copy`, IIRC I ensured that `CategoricalArray` constructors always make a copy. BTW, that means that if you don't need a copy for `refkeycol`, better use `convert` to avoid it if the input is already a `CategoricalArray`.

Reply:
I need a copy as I `droplevels!` in the next line. Done. I add test to check that we actually make a copy.

Reply:
But do you need a copy for `keycol` too?

Reply:
We do (and there was another bug related to this that I have fixed if `keycol` has nonexistent levels).

However, I have a question here - why we want `keycol` to allow `Missing` even if originally it does not have `missing`s? (this is the original design and I respect it, but I do not understand the reason for such an approach)

Reply:
The original design was probably based on `DataArray` everywhere, so it makes sense to change it to return arrays which do not allow for missing where possible.

Reply:
ok. I removed allowing missing if not needed (I hope it is ok - you know `Missings` best 😄):
* I create categorical using `categorical`;
* I do not force `col` in single index case to allow missings;
* I do not force `df1` to allow missings;